build-system/tasks/serve.js

```diff
@@ -161,19 +204,12 @@ async function stopServer() {
 
 /**
  * Closes the existing server and restarts it
+ * @param {ServerOptionsDef=} serverOptions
  */
-async function restartServer() {
+async function restartServer(serverOptions = {}) {
   stopServer();
-  if (argv.new_server) {
-    try {
-      buildNewServer();
-    } catch {
-      log(red('ERROR:'), 'Could not build', cyan('AMP Dev Server'));
-      return;
-    }
-  }
```

Comment:
This is essential here. Why was this moved out? `doServe()` already builds the new server for the startup case.

Reply:
reverted

Reply:
so `doServe` does but the visual-diff code actually calls `startServer` bypassing doServe and the building of the js files. should i switch visualDiff to call `doServe` instead?

Reply:
i went ahead and moved the call to `buildNewServer` from `doServe` to `startServer`. LMK what you think